### PR TITLE
fix(prime-runs): close failed and cancelled evaluations through the status API

### PR DESCRIPTION
**Pull request by kcoopermiller**
## Summary

The evaluations API accepts `PUT /evaluations/{id}` with `status: FAILED | CANCELLED` and an optional `error_message`. prime-runs still recorded a non-completed outcome under `metadata.prime_runs` and warned that the run would keep showing as running, so every crashed or Ctrl-C'd local `uv run eval` stayed RUNNING on the dashboard forever.

`EvalsBackend.finalize` now closes the run through that endpoint:

- Config/summary go up first on their own `PUT`; the terminal status is a second request, so a status guard that rejects the update (409) never takes the run's config with it.
- `failed` → `FAILED`, `cancelled` → `CANCELLED`, `crashed` → `FAILED` with `crashed: <reason>` as the message. `error_message` is capped at the API's 4096 characters.
- A 409 (the platform already closed the run, e.g. a concurrent `/finalize`) is logged at INFO and swallowed, matching the RFT backend. A 400 for a hosted evaluation still raises: attached runs never reach this path because the launcher owns their status.
- The `metadata.prime_runs` marker and the "keep showing as running" warning are removed, along with the README's "Not yet available" section.

---

> [!NOTE]
> **Medium Risk**
> Changes eval run lifecycle and platform API calls on failure paths; mis-handled status updates could still leave incorrect dashboard state, but behavior is narrower and better aligned with the evaluations API.
> 
> **Overview**
> **Failed, cancelled, and crashed local evaluations now close on the platform** instead of staying `RUNNING` forever. `EvalsBackend.finalize` no longer writes terminal state to `metadata.prime_runs` or warns that the dashboard cannot reflect the outcome.
> 
> For non-`COMPLETED` exits it sends **`PUT /evaluations/{id}`** with `FAILED` or `CANCELLED` (crashes map to `FAILED` with a `crashed: …` `error_message`, truncated at 4096 chars). **Config and metrics go in a separate PUT first**, so a failed metadata/metrics update does not block the status close; **HTTP 409** (run already closed) is logged and ignored like RFT.
> 
> README documents this behavior and drops the **“Not yet available”** note about stuck evaluations. Tests assert the new API payloads and edge cases (summary rejection, 409, hosted 400).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5ad820b280766fd0d0a8ee6161860fa51fb24859. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/prime-runs/src/prime_runs/backend.py [hunk 1]
```diff
@@ -142,17 +155,31 @@ def finalize(
...
-        self.update(run_id, config={**(config or {}), "prime_runs": terminal}, summary=summary)
-        logger.warning(
-            "Run %s %s, but the evaluations API cannot record that; it will keep "
-            "showing as running. Recorded the failure in metadata.prime_runs.",
-            run_id,
-            status.value,
-        )
+        # Two requests: the status guard rejects the whole update when the run is
+        # already closed, and the config/summary should land either way.
+        self.update(run_id, config=config, summary=summary)
```
[the remark this answers is not in the document]
Comment (kcoopermiller) [reply packages/prime-runs/src/prime_runs/backend.py]: Valid. Fixed in 5ad820b2: the config/summary update is now wrapped, a rejected or failed request is logged at WARNING, and the terminal status is sent regardless. Test: test_a_rejected_summary_does_not_stop_the_run_from_closing (metrics PUT answers 422, status PUT still lands).

**Review by kennethnym — APPROVED** [review]
